kombu/utils/url.py

```diff
@@ -38,6 +44,24 @@ def parse_url(url):
     # type: (str) -> Dict
     """Parse URL into mapping of components."""
     scheme, host, port, user, password, path, query = _parse_url(url)
+    if query:
+        keys = list(query.keys())
+        for key in keys:
+            if key.startswith('ssl_'):
```

Comment:
We could turn this into a generator comprehension.
The code will be clearer that way.

Reply:
interesting thought, i'll try it out and see how it looks

Reply:
ok I added a list comprehension instead because that cuts down on an extra line, makes the code less nested, and still works around the pitfall of `.keys` being a generator.